lib/pavilion/plugins/commands/show.py

```diff
@@ -390,69 +415,126 @@ def _functions_cmd(self, _, args):
                 title="Available Expression Functions"
             )
 
-    @show_cmd('host')
-    def _hosts_cmd(self, pav_cfg, args):
-        """List all known host files."""
+    def show_vars(self, pav_cfg, config, conf_type):
+
+        file = resolver.TestConfigResolver(pav_cfg)._find_config(conf_type,
+                                                                 config)
+        with file.open() as config_file:
+            config_data = file_format.TestConfigLoader().load(config_file)
+
+        data = []
+        col_names = ['Variables', ' ', 'Type', 'Value']
+
+        name = config
+        path = file
+        config = config_data
+
+        for var in config['variables'].keys():
+            var_out = var
+            index = 0
+            for subvar in config['variables'][var]:
+                if type(subvar) is yaml_config.elements.ConfigDict:
+                    type_out = 'dict'
+                    index_out = index
+                    for key, val in subvar.items():
+                        data.append({
+                            'Variables': var_out,
+                            ' ': index_out,
+                            'Type': type_out,
+                            'Value': key + ": " + val,
+                        })
+                        var_out = ''
+                        type_out = ''
+                        index_out = ' '
+                    index += 1
+                else:
+                    data.append({
+                        'Variables': var_out,
+                        ' ': index,
+                        'Type': type(subvar).__name__,
+                        'Value': subvar,
+                    })
+                    var_out = ''
+                    index += 1
 
```

Comment:
I mostly like what you're doing here, but:
1. Give the index column a name. Even even if it makes the column a little fat.
2. You don't have to capitalize the field names, that happens automatically.
3. I don't like the type column, it's confusing. Every variable has an index and is a 'list' of strings or dict of strings.
4. This doesn't give very readable output for multi-valued variables. More on that below.

Reply:
Or right here. 

For each multi-valued variable, make a separate table for each. You already have the rows, just annotate them with an index column. Add a title that says what the variable name is, etc.

This will be ugly for very large sets of variables. Good. Jen likes to do that, and it's horrendous.